test_launch_ros/test/test_launch_ros/launch_frontend/test_node_frontend.py

```diff
@@ -0,0 +1,116 @@
+# limitations under the License.
+
+"""Example of how to parse an xml."""
+
+import io
+import pathlib
+import textwrap
+
+from launch import LaunchService
+from launch.launch_frontend import Parser
+
+from launch_ros.utilities import evaluate_parameters
+
+import pytest
+
+# Scaping the quote is needed in 'a_string' launch configuration, becuase of how the parser works.
+# TODO(ivanpauno): Check if it's possible to avoid that.
+yaml_params = str(pathlib.Path(__file__).parent / 'params.yaml')
+xml_file = \
+    """\
+    <launch>
+        <let name="a_string" value="\\'[2, 5, 8]\\'"/>
+        <let name="a_list" value="[2, 5, 8]"/>
+        <node package="demo_nodes_py" executable="talker_qos" output="screen" name="my_node" namespace="my_ns" args="--number_of_cycles 1">
+            <param name="param1" value="ads"/>
+            <param name="param_group1">
+                <param name="param_group2">
+                    <param name="param2" value="2"/>
+                </param>
+                <param name="param3" value="2, 5, 8" value-sep=", "/>
+                <param name="param4" value="$(var a_list)"/>
+                <param name="param5" value="$(var a_string)"/>
+            </param>
+            <param from="{}"/>
+            <env name="var" value="1"/>
+        </node>
+    </launch>
+    """.format(yaml_params)  # noqa: E501
+xml_file = textwrap.dedent(xml_file)
+yaml_file = \
+    """\
+    launch:
+        - let:
+            name: 'a_string'
+            value: '\\"[2, 5, 8]\\"'
+        - let:
+            name: 'a_list'
+            value: '[2, 5, 8]'
+        - node:
+            package: demo_nodes_py
+            executable: talker_qos
+            output: screen
+            name: my_node
+            namespace: my_ns
+            args: '--number_of_cycles 1'
+            param:
+                -   name: param1
+                    value: ads
+                -   name: param_group1
+                    param:
+                    -   name: param_group2
+                        param:
+                        -   name: param2
+                            value: 2
+                    -   name: param3
+                        value: [2, 5, 8]
+                    -   name: param4
+                        value: $(var a_list)
+                    -   name: param5
+                        value: $(var a_string)
+                -   from: {}
+            env:
+                -   name: var
+                    value: '1'
+    """.format(yaml_params)  # noqa: E501
+yaml_file = textwrap.dedent(yaml_file)
+
+
+@pytest.mark.parametrize('file', (xml_file, yaml_file))
+def test_node_frontend(file):
+    """Parse node xml example."""
+    root_entity, parser = Parser.load(io.StringIO(file))
+    ld = parser.parse_description(root_entity)
+    ls = LaunchService()
+    ls.include_launch_description(ld)
+    assert(0 == ls.run())
+    lc = ls.context
+    evaluated_parameters = evaluate_parameters(
+        lc,
```

Comment:
@ivanpauno nit: using `ls.context` instead of `lc` is not that much longer and makes code much more readable.